client/packages/common/src/ui/forms/JsonForms/components/Text.tsx

```diff
@@ -7,13 +7,17 @@ export const stringTester = rankWith(3, schemaTypeIs('string'));
 
 const UIComponent = (props: ControlProps) => {
   const { data, handleChange, label, path } = props;
+  const [localData, setLocalData] = useState<string>(data);
   return (
     <DetailInputWithLabelRow
       label={label}
       inputProps={{
-        value: data,
+        value: localData,
         sx: { margin: 0.5, width: '100%' },
-        onChange: e => handleChange(path, e.target.value),
+        onChange: e => {
+          setLocalData(e.target.value);
+        },
+        onBlur: () => handleChange(path, localData),
```

Comment:
To prevent unnecessary re-renders of the entire Form structure on every keystroke, as mentioned in [original post](https://github.com/openmsupply/open-msupply/pull/173#issue-1279465955)